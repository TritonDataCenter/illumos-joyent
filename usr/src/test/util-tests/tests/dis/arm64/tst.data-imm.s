/*
 * This file and its contents are supplied under the terms of the
 * Common Development and Distribution License ("CDDL"), version 1.0.
 * You may only use this file in accordance with the terms of version
 * 1.0 of the CDDL.
 *
 * A full copy of the text of the CDDL should have accompanied this
 * source.  A copy of the CDDL is also available via the Internet at
 * http://www.illumos.org/license/CDDL.
 */

/*
 * Copyright 2018 Joyent, Inc.
 */

/*
 * Test Data - Immediate group dissassembly
 */

.text
.align 16
.globl libdis_test
.type libdis_test, @function
libdis_test:
	/* ADR{P} */
	ADR XZR, branch_target
	ADRP X0, libdis_test

	/* ADD/SUB - Immediate */
	ADD W0, WSP, #4
	ADD WSP, W19, #1, LSL #12
	ADDS X10, X1, #255
	ADDS X10, SP, #255
	SUB SP, X1, #16, LSL #12
	SUBS X0, SP, #8

	/*TODO: Logical - immediate */

	/* Move wide - immediate */
	MOVN W4, #65000, LSL #16
	MOVN X15, #255, LSL #48
	MOVZ W4, #1
	MOVZ X13, #2, LSL #32
	MOVK WZR, #16
	MOVK X0, #4, LSL #16

	/* Bitfield - immediate */
	SBFM W4, W9, #0, #0
	SBFM X3, X10, #1, #9
	BFM W4, W9, #0, #31
	BFM X10, XZR, #2, #57
	UBFM W1, W9, #23, #3
	UBFM XZR, X0, #0, #6

	/* Extract */
	EXTR W0, W1, WZR, #31
	EXTR W0, WZR, W1, #4
	EXTR WZR, W5, W30, #1
	EXTR X11, X1, XZR, #60
	EXTR X18, XZR, X19, #43
	EXTR XZR, X25, X30, #3

	branch_target: ADD X0, X0, X0

.size libdis_test, [.-libdis_test]
