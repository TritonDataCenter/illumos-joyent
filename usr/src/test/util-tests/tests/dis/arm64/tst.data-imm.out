    libdis_test:      5f 03 00 10  adr xzr, +0x68 	<libdis_test+0x68>
    libdis_test+0x4:  00 00 00 90  adrp x0, -0x4 	<libdis_test>
    libdis_test+0x8:  e0 13 00 11  add w0, wsp, #0x4
    libdis_test+0xc:  7f 06 40 11  add wsp, w19, #0x1, lsl #12
    libdis_test+0x10: 2a fc 03 b1  adds x10, x1, #0xff
    libdis_test+0x14: ea ff 03 b1  adds x10, sp, #0xff
    libdis_test+0x18: 3f 40 40 d1  sub sp, x1, #0x10, lsl #12
    libdis_test+0x1c: e0 23 00 f1  subs x0, sp, #0x8
    libdis_test+0x20: 04 bd bf 12  movn w4, #0xfde8, lsl #16
    libdis_test+0x24: ef 1f e0 92  movn x15, #0xff, lsl #48
    libdis_test+0x28: 24 00 80 52  movz w4, #0x1
    libdis_test+0x2c: 4d 00 c0 d2  movz x13, #0x2, lsl #32
    libdis_test+0x30: 1f 02 80 72  movk wzr, #0x10
    libdis_test+0x34: 80 00 a0 f2  movk x0, #0x4, lsl #16
    libdis_test+0x38: 24 01 00 13  sbfm w4, w9, #0x0, #0x0
    libdis_test+0x3c: 43 25 41 93  sbfm x3, x10, #0x1, #0x9
    libdis_test+0x40: 24 7d 00 33  bfm w4, w9, #0x0, #0x1f
    libdis_test+0x44: ea e7 42 b3  bfm x10, xzr, #0x2, #0x39
    libdis_test+0x48: 21 0d 17 53  ubfm w1, w9, #0x17, #0x3
    libdis_test+0x4c: 1f 18 40 d3  ubfm xzr, x0, #0x0, #0x6
    libdis_test+0x50: 20 7c 9f 13  extr w0, w1, wzr, #0x1f
    libdis_test+0x54: e0 13 81 13  extr w0, wzr, w1, #0x4
    libdis_test+0x58: bf 04 9e 13  extr wzr, w5, w30, #0x1
    libdis_test+0x5c: 2b f0 df 93  extr x11, x1, xzr, #0x3c
    libdis_test+0x60: f2 af d3 93  extr x18, xzr, x19, #0x2b
    libdis_test+0x64: 3f 0f de 93  extr xzr, x25, x30, #0x3
    libdis_test+0x68: 00 00 00 8b  add x0, x0, x0
